project/Scope.rst

```diff
@@ -0,0 +1,23 @@
+Scope of Halium
+===============
+
+Halium's scope primarily consists of 2 parts:
+
+1. Halium's Android is a minimal Android distribution designed to run within an LXC container. It will provide a set of interfaces to let the host of LXC container utilize Android's HAL to communicate with the hardware of Android-based smartphones. These interfaces come in 3 forms:
+
+    1. Native Android interfaces. (e.g. rild's socket, Audio HAL (dlopen'ed directly), (original) WiFi HAL (a standard libhybris wrapper))
+    2. Libhybris compat layer. Usually used to wrap Android C++ interfaces, contains some logic, and has its counter-part compiled in the Android tree. (e.g. HWC2 compat layer)
+    3. Custom interfaces. This will contain more than a compat layer and may have quite deep integration within Android code. (e.g. Ubuntu Touch's camera interfaces, which consists of camera compat layer, custom media compat layer, and a deep modification of Android's libmedia). If possible, we would like to reduce this kind of interface as much as possible to reduce our workload, especially those doing deep modification to Android.
```

Comment:
Might you be able to link to examples of any of these? They would be helpful to illustrate your ideas

Reply:
Please see if the links I added are OK.